practice/earthquake/site.js

```diff
@@ -0,0 +1,169 @@
+'use strict';
+/*
+This program creates a world map and adds tokens for each earthquake in the past
+7 days. The tokens vary in size based on magnitude of the earthquake (larger
+token, larger magnitude), and in opacity based on age (older earthquake: more
+opaque).
+It uses OpenLayers to create the map and map objects. The EarthquakeData is
+sorurced from usgs.gov.
+ */
+
+/**
+ * Fixes linter issues with OpenLayers.
+ */
+if (!window.ol) {
+  var ol = {};
+}
+
+var ACCESS_TIME_UTC = new Date().getTime();
+
+
+/**
+ * Inputs new Earthquake Data from usgs.gov.
+ */
+function getEarthquakeData() {
+  var url = 'http://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_week.geojson';
+  return Promise.resolve($.ajax({
+    dataType: 'json',
+    url: url
+  }));
+}
+
+/**
+ * Data Transformation- Pulls magnitude, longitude, latitude, and time of each
+ * earthquake into an array and maps them together into an array of arrays.
+ * @param  {JSON object} earthquakeData: geoJSON object containing all
+ * earthquakeData for the previous week.
+ * @return {Array}                array of arrays containing magnitude,
+ *                                 coordinates, and time of each earthquake.
+ */
+function getEqCoordsMagsAndTimes(earthquakeData) {
+  var features = earthquakeData.features;
+  return _.map(features, function(feature) {
+    var magnitude = feature.properties.mag;
+    var longitude = feature.geometry.coordinates[0];
+    var latitude = feature.geometry.coordinates[1];
+    var time = feature.properties.time;
+    return [magnitude, longitude, latitude, time];
+  });
+}
```

Comment:
I think this data transformation makes the resulting code harder to read. It's taking nicely named properties and then removing their names and putting them in an array indexed with magic numbers. Even if most of the properties on `features` you don't care about in rendering, the rendering code will be much easier to read with those names.